### DISCUSSION TITLE
mongodb appender - how to handle random classes not having to_bson?

### DISCUSSION THREAD

**Discussion by neongrau** [reidmorrison/semantic_logger#270] [Q&A]
Fails to log because it bails even at a simple error `Aws::S3::Errors::NoSuchBucket`.
```
SemanticLogger::Appender::MongoDB -- Exception: BSON::Error::UnserializableClass: 
Value does not define its BSON serialized type: The specified bucket does not exist
```

Is there a simple way to just bson serialize anything not responding to `to_bson` as `.to_s` or `.inspect` ?

**Comment by reidmorrison** [comment]
Sounds like a good idea, want to provide a pull request, checking if the content supports `to_bson` before trying to serialize it for the MongoDB appender?
